railties/lib/rails/generators/rails/app/templates/Dockerfile.tt

```diff
@@ -33,6 +33,12 @@ ENV RAILS_LOG_TO_STDOUT="1" \
 COPY Gemfile Gemfile.lock ./
 RUN bundle install
 
+<% if using_node? -%>
+# install node modules
+COPY yarn.* ./
+RUN yarn install
```

Comment:
This is already happening as part of the `assets:precompile`. Are you separating it out as a layer optimization?

Reply:
I removed this for now.  Yes, a `yarn install` without any caching whatsoever is rather time consuming, and not exactly an ideal user experience.  And there is a range of options from "I don't care how long it takes, I just want my Dockerfile to be gorgeous"  to "I don't care how complicated my Dockerfile is, I just want deploys to be fast".

I explore these options on my Rails cookbooks page.  Picking one approach now is premature.  Let's discuss: I don't care whether the discussion happens in git issues, Rails Discourse, Fly's community forum or elsewhere.  I'm flexible.

Reply:
`assets:precompile` is run as part of image creation, so I believe it's the same experience. It's all this done at the image stage. Don't think there's a cache difference?